### PR TITLE
aks-preview: support specifying custom http headers in aks upgrade

### PR DESCRIPTION
**Pull request by nilo19**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by nilo19** [comment]
@miwithro @robbiezhang @zqingqing1 can I get a review?

**Review by miwithro — APPROVED** [review]

File: src/aks-preview/azext_aks_preview/_help.py [hunk 1]
```diff
@@ -387,6 +387,9 @@
         - name: --node-image-only
           type: bool
           short-summary: Only upgrade node image for agent pools.
+        - name: --aks-custom-headers
```
Comment (yungezz) [inline src/aks-preview/azext_aks_preview/_help.py:390]: how about just `--custom-headers`? the command group name is aks already, aks is redundant

File: src/aks-preview/azext_aks_preview/custom.py [hunk 2]
```diff
@@ -2056,7 +2057,9 @@ def aks_upgrade(cmd,    # pylint: disable=unused-argument, too-many-return-state
     instance.service_principal_profile = None
     instance.aad_profile = None
 
-    return sdk_no_wait(no_wait, client.create_or_update, resource_group_name, name, instance)
+    headers = get_aks_custom_headers(aks_custom_headers)
```
Comment (yungezz) [inline src/aks-preview/azext_aks_preview/custom.py:2060]: any validation, dedup on input header values?

Comment (nilo19) [reply src/aks-preview/azext_aks_preview/custom.py:2060]: the validation is in get_aks_custom_headers

File: src/aks-preview/azext_aks_preview/_help.py (hunk 1 above)
Comment (nilo19) [reply src/aks-preview/azext_aks_preview/_help.py:390]: just to keep aligned with the `aks-custom-headers` in other commands like `az aks create`

**Review by yungezz — APPROVED** [review]

**Comment by nilo19** [comment]
@fengzhou-msft can we merge it?

**Pull request merged**